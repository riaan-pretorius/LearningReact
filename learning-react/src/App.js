import React, { Component } from "react";
import Home from "./pages/Home";
import GitHubViewer from "./pages/GitHubViewer";
import AboutUs from "./pages/AboutUs";
import ContactUs from "./pages/ContactUs";
import Toolbar from "./components/Toolbar";
import { Container, Row, Col } from "reactstrap";
import { BrowserRouter as Router, Route } from "react-router-dom";

import "./css/bootstrap.css";

const links = [
  { to: "/", text: "Home" },
  { to: "/githubviewer", text: "GitHub Viewer" },
  { to: "/about", text: "About" },
  { to: "/contact", text: "Contact" }
];

class App extends Component {
  render() {
    return (
      <Container style={styles.container}>
        <Row>
          <Col>
            <Router>
              <div>
                <Toolbar title="Project name" links={links} />
                <hr />
                <Route exact path="/" component={Home} />
                <Route path="/githubviewer" component={GitHubViewer} />
                <Route path="/about" component={AboutUs} />
                <Route path="/contact" component={ContactUs} />
              </div>
            </Router>
          </Col>
        </Row>
      </Container>
    );
  }
}
const styles = {
  container: {
    paddingTop: 20
  }
};

export default App;
