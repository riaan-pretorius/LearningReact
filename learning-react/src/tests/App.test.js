import React from 'react'
import App from '../App'
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'

test('App Test Snapshot', () => {
  const component = renderer.create(
    <App />
  )
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});

test('App Test', () => {
  const app = mount(
    <App />
  )
  
  expect(app.find('.container').exists()).toBeTruthy()
  expect(app.find('.row').exists()).toBeTruthy()
  expect(app.find('.col').exists()).toBeTruthy()
})
