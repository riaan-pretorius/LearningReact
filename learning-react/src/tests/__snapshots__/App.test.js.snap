// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Snapshot 1`] = `
<div
  className="container"
  style={
    Object {
      "paddingTop": 20,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        <div
          className="col"
        >
          <div
            className="row"
          >
            <div
              className="col col-sm-4"
            >
              <h3
                className="text-muted"
              >
                Project name
              </h3>
            </div>
            <div
              className="col col-sm-8"
            >
              <ul
                className="nav nav-pills"
                style={
                  Object {
                    "float": "right",
                  }
                }
              >
                <li
                  className="nav-item"
                >
                  <a
                    aria-current={false}
                    className="nav-link"
                    href="/"
                    onClick={[Function]}
                    style={undefined}
                  >
                    Home
                  </a>
                </li>
                <li
                  className="nav-item"
                >
                  <a
                    aria-current={false}
                    className="nav-link"
                    href="/githubviewer"
                    onClick={[Function]}
                    style={undefined}
                  >
                    GitHub Viewer
                  </a>
                </li>
                <li
                  className="nav-item"
                >
                  <a
                    aria-current={false}
                    className="nav-link"
                    href="/about"
                    onClick={[Function]}
                    style={undefined}
                  >
                    About
                  </a>
                </li>
                <li
                  className="nav-item"
                >
                  <a
                    aria-current={false}
                    className="nav-link"
                    href="/contact"
                    onClick={[Function]}
                    style={undefined}
                  >
                    Contact
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
        <hr />
      </div>
    </div>
  </div>
</div>
`;
